src/internal/middleware/logging/interceptor.go

```diff
@@ -437,6 +438,12 @@ func (li *LoggingInterceptor) UnaryServerInterceptor(ctx context.Context, req in
 	service, method := parseMethod(info.FullMethod)
 	ctx = getCommonLogger(ctx, service, method)
 
+	md, ok := metadata.FromIncomingContext(ctx)
```

Comment:
This only handles unary requests; most of our requests are streaming, so you'll need to log there as well.   Do keep in mind that we already log the command with every message logged during a request (line 368).  (The common logger is used for every line the RPC handler logs, including the request and response.  The request logger logs the request; the response logger logs the response.)  If there needs to be a log line that contains "audit log" with the text of the command, I'd do it in getCommonLogger.  Using the context passed to that function, extract the request id (to tie the audit line to the rest of the request), username, and command.  then `log.Info(ctx, "pachctl command: " + command, zap.String("x-request-id", requestID), zap.String("user", user))`.  It all seems a little redundant to me; it's an extra log line with no additional information.

Reply:
The audit log line is to provide the users a better way to list all the pachctl commands that have been run against their cluster. At the moment it's difficult to obtain this list from the logs because there are so many log lines printed for just one command. We could parse out the command field from the other log lines and then group them into unique pachctl commands for an audit log but that is a lot more computation. Eventually, I think we should remove this line from the logs and add it as an event log. 

Moving this to getCommonLogger makes sense to me

Reply:
Fair enough.  Yeah, basically treat it as an unrelated log in getCommonLogger, I think that will work fine.